test/jasmine/karma.conf.js

```diff
@@ -1,30 +1,102 @@
-var testFileGlob = arg ? arg : 'tests/*_test.js';
-var isSingleSuiteRun = (arg && arg.indexOf('bundle_tests/') === -1);
-var isRequireJSTest = (arg && arg.indexOf('bundle_tests/requirejs') !== -1);
-var isIE9Test = (arg && arg.indexOf('bundle_tests/ie9') !== -1);
+var argv = minimist(process.argv.slice(4), {
+    string: ['bundleTest', 'width', 'height'],
+    'boolean': ['info', 'watch', 'verbose', 'Chrome', 'Firefox'],
+    alias: {
+        'Chrome': 'chrome',
+        'Firefox': ['firefox', 'FF'],
+        'bundleTest': ['bundletest', 'bundle_test']
+    },
+    'default': {
+        info: false,
+        watch: false,
+        verbose: false,
+        width: '1035',
+        height: '617'
+    }
+});
+
+if(argv.info) {
+    console.log([
+        'plotly.js karma runner for jasmine tests CLI info',
+        '',
+        'Examples:',
+        '',
+        'Run `axes_test.js`, `bar_test.js` and `scatter_test.js` suites in `autoWatch` / multiple run mode:',
+        '  $ npm run test-jasmine -- axes bar_test.js scatter --watch',
+        '',
+        'Run all tests with the `noCI` tag on Firefox in a 1500px wide window:',
+        '  $ npm run test-jasmine -- --tags=noCI --FF --width=1500',
+        '',
+        'Run the `ie9_test.js` bundle test with the verbose reporter:',
+        '  $ npm run test-jasmine -- --bundleTest=ie9 --verbose',
+        '',
+        'Arguments:',
+        '  - All non-flagged arguments corresponds to the test suites in `test/jasmine/tests/` to be run.',
+        '    No need to add the `_test.js` suffix, we expand them correctly here.',
+        '  - `--bundleTest` set the bundle test suite `test/jasmine/bundle_tests/ to be run.',
+        '    Note that only one bundle test can be run at a time.',
+        '',
+        'Other options:',
+        '  - `--info`: show this info message',
+        '  - `--Chrome` (alias `--chrome`): run test in (our custom) Chrome browser',
+        '  - `--Firefox` (alias `--FF`, `--firefox`): run test in (our custom) Firefox browser',
+        '  - `--watch (dflt: `false`)`: run karma in `autoWatch` / multiple run mode',
+        '  - `--verbose` (dflt: `false`): show test result using verbose reporter',
+        '  - `--tags`: run only test with given tags (using the `jasmine-spec-tags` framework)',
+        '  - `--width`(dflt: 1035): set width of the browser window',
+        '  - `--height` (dflt: 617): set height of the browser window',
+        '',
+        'For info on the karma CLI options run `npm run test-jasmine -- --help`'
+    ].join('\n'));
+    process.exit(0);
+}
+
+var SUFFIX = '_test.js';
+var basename = function(s) { return path.basename(s, SUFFIX); };
+var merge = function(_) {
+    var list = [];
+
+    (Array.isArray(_) ? _ : [_]).forEach(function(p) {
+        list = list.concat(p.split(','));
+    });
+
+    return list;
+};
+var glob = function(_) {
+    return _.length === 1 ?
+        _[0] + SUFFIX :
+        '{' + _.join(',') + '}' + SUFFIX;
+};
```

Comment:
This logic :arrow_heading_up: is to normalize the test file name inputs, so that:

```
npm run test-jasmine -- axes bar scatter
```

does the same as

```
npm run test-jasmine -- axes_test.js tests/scatter_test.js bar
```

I hope you enjoy not having to type those `tests/` and `_test.js` to run individual test suite.